src/bp/ui-studio/src/web/views/OneFlow/diagram/nodes/PromptContents/index.tsx

```diff
@@ -49,17 +49,19 @@ const PromptContents: FC<Props> = ({ node, selectedNodeItem, currentLang }) => {
           <span className={style.content}>{params?.output && `$${params?.output}`}</span>
         )}
       </div>
-      {next?.map((item, i) => {
-        const outputPortName = `out${i}`
-        return (
-          <div key={`${i}.${item}`} className={style.contentWrapper}>
-            <div className={cx(style.content, style.readOnly)}>
-              {lang.tr(item.caption)}
-              <StandardPortWidget name={outputPortName} node={node} className={style.outRouting} />
+      {next
+        ?.filter(x => params.cancellable || x.condition !== 'thisNode.cancelled === true')
+        .map((item, i) => {
+          const outputPortName = `out${i}`
+          return (
+            <div key={`${i}.${item}`} className={style.contentWrapper}>
+              <div className={cx(style.content, style.readOnly)}>
+                {lang.tr(item.caption)}
+                <StandardPortWidget name={outputPortName} node={node} className={style.outRouting} />
+              </div>
             </div>
-          </div>
-        )
-      })}
```

Comment:
I'm not sure about this piece of code. It only hides the transition UI-side, but it is still stored & processed by the backend. Feels like some magic hidden somewhere.

Ideally when you enable "cancellable" it would add the transition , and remove it when you uncheck it